exams/pearson/audit_test.py

```diff
@@ -0,0 +1,281 @@
+    mocker.patch.object(s3_store, 'get_bucket', return_value='.the.bucket.')
+    key = s3_store.get_s3_key('filename', 'filetype')
+    assert key.bucket == s3_store.get_bucket.return_value
+    assert key.key == 'exam_audits/filetype/filename'
+    s3_store.get_bucket.assert_called_once_with()
+
+
+def test_exam_data_auditor_fingerprint(auditor):
+    """Test that fingerprint strips whitespace from the key"""
+    assert auditor.fingerprint == FINGERPRINT_STRIPPED
+
+
+def test_exam_data_auditor_public_key(auditor):
+    """Test that this decodes a base64 encoded pubkey"""
+    assert auditor.public_key == PUBLIC_KEY
+
+
+def test_exam_data_auditor_configure(auditor):
+    """Test that configure() succeeds with correct settings"""
+    auditor.configure()
+    assert auditor.gpg.list_keys().fingerprints == [FINGERPRINT_STRIPPED]
+    assert auditor.configured is True
+
+
+def test_exam_data_auditor_configure_already_configured(auditor, mocker):
+    """Test that configure() succeeds if already configured"""
+    mocker.patch.object(auditor, 'get_fingerprints', return_value=[FINGERPRINT_STRIPPED])
+    mocker.spy(auditor.gpg, 'import_keys')
+    auditor.configure()
+    assert auditor.gpg.import_keys.call_count == 0
+    assert auditor.configured is True
+
+
+def test_exam_data_auditor_enabled(auditor, missing_settings, mocker):
+    """Test that audit() respected the enabled flag"""
+    mocker.patch.object(auditor, 'configure')
+    mocker.patch.object(auditor, 'upload_encrypted_file')
+    # auditor is disabled
+    with override_settings(**{
+        'EXAMS_AUDIT_ENABLED': False,
+    }):
+        auditor.audit('file', 'filetype')
+    assert auditor.configure.call_count == 0
+    # auditor is enabled
+    with override_settings(**{
+        'EXAMS_AUDIT_ENABLED': True,
+    }):
+        auditor.audit('file', 'filetype')
+    assert auditor.configure.call_count == 1
+
+
+def test_exam_data_auditor_configure_missing_settings(auditor, missing_settings, mocker):
+    """Test that configure() fails with missing settings"""
+    mocker.spy(auditor, 'get_fingerprints')
+    with pytest.raises(ImproperlyConfigured):
+        auditor.configure()
+    assert auditor.get_fingerprints.call_count == 0
+    assert auditor.configured is False
+
+
+def test_exam_data_auditor_configure_bad_settings(auditor, invalid_settings):
+    """Test that configure() fails with invalid settings"""
+    with pytest.raises(ImproperlyConfigured):
+        auditor.configure()
+    assert auditor.configured is False
+
+
+def test_exam_data_auditor_configure_not_imported(auditor, mocker):
+    """Test that configure() succeeds if already configured"""
+    mock = mocker.patch.object(auditor.gpg, 'import_keys')
+    mock.return_value.counts = {
+        'not_imported': 1,
+    }
+    with pytest.raises(ImproperlyConfigured):
+        auditor.configure()
+    assert auditor.configured is False
+
+
+def test_exam_data_s3_store_upload(s3_store, mocker):
```

Comment:
I think this test case is confusing me the most out of all of them, so I think it should have a docstring 😄 . Also, it seems like this is a straightforward test of `exams.pearson.audit.S3AuditStorage#upload` and not a test for any `ExamDataAuditor` functionality. If that's true, can you move this to be nearer to the other `S3AuditStorage`-specific test cases?